haystack/components/preprocessors/csv_document_splitter.py

```diff
@@ -13,18 +13,24 @@
 
 logger = logging.getLogger(__name__)
 
+SplitMode = Literal["threshold", "row-wise"]
+
 
 @component
 class CSVDocumentSplitter:
     """
-    A component for splitting CSV documents into sub-tables based on empty rows and columns.
+    A component for splitting CSV documents into sub-tables based on split arguments.
 
-    The splitter identifies consecutive empty rows or columns that exceed a given threshold
+    The splitter supports two modes of operation:
+    - identify consecutive empty rows or columns that exceed a given threshold
     and uses them as delimiters to segment the document into smaller tables.
+    - split each row into a separate sub-table, represented as a Document.
+
     """
 
     def __init__(
         self,
+        split_mode: SplitMode = "threshold",
```

Comment:
We can't add this here, it needs to be last in it the init args otherwise this is a breaking change if anyone initialized this using positional args instead of keyword arguments